### PR TITLE
Helper: Review all functions

### PR DESCRIPTION
**Pull request by 1480c1**
Currently looking through the helper script to see if there is anything that can be improved etc.

Most of this is from shellcheck. Going to test once my windows vm actually starts up

Hoping to do compile.sh afterwards. maybe.

### CODE REVIEW

File: build/media-suite_helper.sh [hunk 1]
```diff
@@ -661,15 +661,8 @@ do_pkgConfig() {
 do_readoptionsfile() {
     local filename="$1"
     if [[ -f "$filename" ]]; then
-        < $filename dos2unix |
```
Comment (wiiaboo) [inline build/media-suite_helper.sh]: Was this just removing the comments? If so, revert it. The comments are useful as documentation for how to format the options files.

File: build/media-suite_helper.sh [hunk 2]
```diff
@@ -769,7 +769,7 @@ do_changeFFmpegConfig() {
     eval "$(sed -n '/EXTERNAL_LIBRARY_VERSION3_LIST=/,/^"/p' "$config_script")"
 
     # handle gpl libs
-    local gpl=(${EXTERNAL_LIBRARY_GPL_LIST//_/-} gpl)
+    local gpl=("${EXTERNAL_LIBRARY_GPL_LIST//_/-}" gpl)
```
Comment (wiiaboo) [inline build/media-suite_helper.sh]: Doesn't this make all the non-gpl libs into a single entry in the array?

File: build/media-suite_helper.sh (hunk 1 above)
Comment (1480c1) [reply build/media-suite_helper.sh]: It was mainly for removing the dos2unix. Reverted

File: build/media-suite_helper.sh [hunk 3]
```diff
@@ -1113,14 +1113,14 @@ do_cmake() {
...
-    [[ -z "$skip_build_dir" ]] && create_build_dir $cmake_build_dir
+    [[ -z "$skip_build_dir" ]] && create_build_dir "$cmake_build_dir"
     extra_script pre cmake
     [[ -f "$(get_first_subdir)/do_not_reconfigure" ]] &&
         return
     log "cmake" cmake "$root" -G Ninja -DBUILD_SHARED_LIBS=off \
         -DCMAKE_TOOLCHAIN_FILE="$LOCALDESTDIR/etc/toolchain.cmake" \
         -DCMAKE_INSTALL_PREFIX="$LOCALDESTDIR" -DUNIX=on \
-        -DCMAKE_BUILD_TYPE=Release $bindir ${cmake_extras[@]} "$@"
+        -DCMAKE_BUILD_TYPE=Release "$bindir" "${cmake_extras[@]}" "$@"
```
Comment (wiiaboo) [inline build/media-suite_helper.sh]: `bindir` might be empty. Don't quote this.

File: build/media-suite_helper.sh (hunk 2 above)
Comment (1480c1) [reply build/media-suite_helper.sh]: Need to check again for that, probably yes, need to fix

File: build/media-suite_helper.sh [hunk 4]
```diff
@@ -1154,12 +1154,11 @@ do_meson() {
...
     shift 1
 
-    local PKG_CONFIG=pkg-config
     create_build_dir
     extra_script pre meson
-    CC=gcc CXX=g++ \
+    PKG_CONFIG=pkg-config CC=gcc CXX=g++ \
         log "meson" meson "$root" --default-library=static --buildtype=release \
-        --prefix="$LOCALDESTDIR" --backend=ninja $bindir "$@"
+        --prefix="$LOCALDESTDIR" --backend=ninja "$bindir" "$@"
```
Comment (wiiaboo) [inline build/media-suite_helper.sh]: Same `bindir` might be empty.

File: build/media-suite_helper.sh [hunk 5]
```diff
@@ -1451,7 +1451,7 @@ do_autoreconf() {
         -z "$(ls "$basedir"/build_successful* 2> /dev/null)" ]]; } ||
         [[ ! -f configure ]]; then
         extra_script pre autoreconf
-        log "autoreconf" autoreconf -fiv $*
+        log "autoreconf" autoreconf -fiv "$@"
```
Comment (wiiaboo) [inline build/media-suite_helper.sh]: As long as this doesn't break with no parameters passed.

File: build/media-suite_helper.sh (hunk 2 above)
Comment (1480c1) [reply build/media-suite_helper.sh]: Found a way to use read and properly set the arrays. Going to rerun the suite to test. I sort of wish there was a way to quickly run the suite with all configs to test.

**Comment by wiiaboo** [comment]
Since most of these changes are on helper functions, you could just test one or two cases for regressions for each and see if anything's broken.

File: build/media-suite_helper.sh (hunk 5 above)
Comment (1480c1) [reply build/media-suite_helper.sh]: As far as I saw, we have not used do_autoreconf with any params and the only one with params for autogen is fontconfig

**Comment by 1480c1** [comment]
Actually, I'm going to work on trying to split this into multiple PRs since some of these are an easy review and some are a bit more annoying logically

File: build/media-suite_helper.sh [hunk 6]
```diff
@@ -64,12 +65,12 @@ set_title() {
...
 
 do_exit_prompt() {
-    if [[ -n $build32 || -n $build64 ]]; then
+    if [[ -n "$build32" || -n "$build64" ]]; then
         create_diagnostic
         zip_logs
     fi
     do_prompt "$*"
-    [[ -n $build32 || -n $build64 ]] && exit 1
+    [[ -n "$build32" || -n "$build64" ]] && exit 1
```
Comment (1480c1) [inline build/media-suite_helper.sh]: Should I just do `-n "${build32}${build64}"`?

Comment (wiiaboo) [reply build/media-suite_helper.sh]: Sure. It's a safety for running the commands on mintty on its own.

**Comment by wiiaboo** [comment]
Helper: trivial quotes (cbdb236cffa88bb3ae59ab182dee8f07c2401d66) causes a regression in
```bash
. /build/media_suite_helper.sh
ffmpegChoice=n do_getFFmpegConfig nonfree; printf '"%s"\n' "${FFMPEG_OPTS[@]}"
```

```
Imported default FFmpeg options from .bat
License: nonfree
"(--disable-autodetect --enable-amf --enable-bzlib --enable-cuda --enable-cuvid --enable-d3d11va --enable-dxva2 --enable-iconv --enable-lzma --enable-nvenc --enable-schannel --enable-zlib --enable-sdl2 --disable-debug --enable-ffnvcodec --enable-nvdec --enable-gmp --enable-libmp3lame --enable-libopus --enable-libvorbis --enable-libvpx --enable-libx264 --enable-libx265 --enable-libdav1d)"
"--enable-amf"
"--enable-bzlib"
"--enable-cuda"
"--enable-cuvid"
"--enable-d3d11va"
"--enable-dxva2"
"--enable-iconv"
"--enable-lzma"
"--enable-nvenc"
"--enable-schannel"
"--enable-zlib"
"--enable-sdl2"
"--disable-debug"
"--enable-ffnvcodec"
"--enable-nvdec"
"--enable-gmp"
"--enable-libmp3lame"
"--enable-libopus"
"--enable-libvorbis"
"--enable-libvpx"
"--enable-libx264"
"--enable-libx265"
"--enable-libdav1d"
```
should be 
```
Imported default FFmpeg options from .bat
License: nonfree
"--disable-autodetect"
"--enable-amf"
"--enable-bzlib"
"--enable-cuda"
"--enable-cuvid"
"--enable-d3d11va"
"--enable-dxva2"
"--enable-iconv"
"--enable-lzma"
"--enable-nvenc"
"--enable-schannel"
"--enable-zlib"
"--enable-sdl2"
"--disable-debug"
"--enable-ffnvcodec"
"--enable-nvdec"
"--enable-gmp"
"--enable-libmp3lame"
"--enable-libopus"
"--enable-libvorbis"
"--enable-libvpx"
"--enable-libx264"
"--enable-libx265"
"--enable-libdav1d"
```

**Comment by wiiaboo** [comment]
I've rebased and merged stuff until "get_external_opts: separate local and setting of pkgname" which is where I stopped reviewing before dinner.

**Comment by wiiaboo** [comment]
Actually, the rest was simple to review while waiting for the pizza, so I've finished it meanwhile. I'm leaving this open in case you still have more of these fixes to add.

**Comment by wiiaboo** [comment]
The fix to the regression was reverting https://github.com/jb-alvarado/media-autobuild_suite/pull/1288/commits/cbdb236cffa88bb3ae59ab182dee8f07c2401d66#diff-1599a266e917d8580ce547a1f1dcc93bR1022, btw.

**Comment by 1480c1** [comment]
None for the helper script. So that's good, now onto the compile script.

**Pull request closed without merging**